kong/clustering/control_plane.lua

```diff
@@ -540,27 +542,34 @@ local function push_config_loop(premature, self, push_config_semaphore, delay)
     if exiting() then
       return
     end
+
     if ok then
-      ok, err = pcall(self.push_config, self)
-      if ok then
-        local sleep_left = delay
-        while sleep_left > 0 do
-          if sleep_left <= 1 then
-            ngx.sleep(sleep_left)
-            break
-          end
+      if next(self.clients) then
+        ok, err = pcall(self.push_config, self)
+        if ok then
+          local sleep_left = delay
+          while sleep_left > 0 do
+            if sleep_left <= 1 then
+              ngx.sleep(sleep_left)
+              break
+            end
+
+            ngx.sleep(1)
 
-          ngx.sleep(1)
+            if exiting() then
+              return
+            end
 
-          if exiting() then
-            return
+            sleep_left = sleep_left - 1
           end
 
-          sleep_left = sleep_left - 1
+        else
+          ngx_log(ngx_ERR, _log_prefix, "export and pushing config failed: ", err)
         end
 
       else
-        ngx_log(ngx_ERR, _log_prefix, "export and pushing config failed: ", err)
+        -- no clients connected, clear the stale config cache
+        self.deflated_reconfigure_payload = nil
```

Comment:
We might not want to clear the cached payload in this case. According to 310722659eead1a6aa8a92a6cf025b5708bb0698, one of the reasons we keep this cache is so that the CP can serve a stale config payload in the event of a database outage.

Reply:
Here's how I wound up dealing with this in my branch:
https://github.com/Kong/kong/blob/c08ee7f03ee15a01a06a85b8df07b60fb5508334/kong/clustering/control_plane.lua#L359-L365

Reply:
This has been changed, please give it another look.